jobclient/python/cook/scheduler/client/jobs.py

```diff
@@ -112,197 +104,181 @@ class Job:
-        :param user: User running this job.
-        :type user: Optional[str], optional
+        :type progress_regex_string: str
         :param datasets: Datasets associated with this job.
-        :type datasets: Optional[list], optional
+        :type datasets: List[Dataset]
         :param gpus: The number of GPUs to request from Cook.
-        :type gpus: Optional[float], optional
-        :param framework_id: Framework ID for this job.
-        :type framework_id: Optional[str], optional
+        :type gpus: int
         :param ports: Ports for this job.
-        :type ports: Optional[int], optional
-        :param submit_time: The UNIX timestamp when this job was submitted.
-        :type submit_time: Optional[int], optional
-        :param retries_remaining: The number of retries remaining for this job.
-        :type retries_remaining: Optional[int], optional
+        :type ports: int
         """
         self.command = command
         self.mem = mem
         self.cpus = cpus
-        self.uuid = uuid if uuid is not None else util.make_temporal_uuid()
+        self.uuid = uuid
         self.name = name
         self.max_retries = max_retries
         self.max_runtime = max_runtime
-        self.state = state
         self.status = status
+        self.state = state
         self.priority = priority
-        self.disable_mea_culpa_retries = disable_mea_culpa_retries
+        self.framework_id = framework_id
+        self.retries_remaining = retries_remaining
+        self.submit_time = submit_time
+        self.user = user
         self.executor = executor
+        self.container = container
+        self.disable_mea_culpa_retries = disable_mea_culpa_retries
         self.expected_runtime = expected_runtime
         self.pool = pool
         self.instances = instances
         self.env = env
         self.uris = uris
-        self.container = container
         self.labels = labels
         self.constraints = constraints
+        self.group = group
         self.groups = groups
         self.application = application
         self.progress_output_file = progress_output_file
         self.progress_regex_string = progress_regex_string
-        self.user = user
         self.datasets = datasets
         self.gpus = gpus
-        self.framework_id = framework_id
         self.ports = ports
-        self.submit_time = submit_time
-        self.retries_remaining = retries_remaining
-
-    def __hash__(self):
-        PRIME = 31
-        result = 1
-        result = (PRIME * result
-                  + hash(self.status) if self.status is not None else 0)
-        result = (PRIME * result
-                  + hash(self.uuid) if self.uuid is not None else 0)
-        return result
-
-    def __eq__(self, other):
-        if self is other:
-            return True
-        if not isinstance(other, self.__class__):
-            return False
-        return self.uuid == other.uuid and self.status == other.status
 
     def to_dict(self) -> dict:
```

Comment:
Where is `Job.to_dict` used?

Reply:
It was mostly a convenience function for showing the entire state of the object at once (something in the vein of `print(json.dumps(job.to_dict(), indent=4))`), although if this were to stay it'd probably be called within an implementation for `__str__`